jobs/integration/conftest.py

```diff
@@ -141,22 +153,58 @@ async def run(self, cmd, *args, stdin=None):
         if hasattr(stdin, "encode"):
             stdin = stdin.encode("utf8")
 
-        stdout, stderr = await proc.communicate(input=stdin)
-        if proc.returncode != 0:
+        stdout, stderr = bytearray(), bytearray()
+
+        def tee(line: bytes, sink: bytearray, fd: int):
+            sink += line
+            write = _tee == "out" and fd == 1
+            write |= _tee == "err" and fd == 2
+            if write or _tee is True:
+                os.write(fd, line)
+
+        async def _read_stream(stream, callback):
+            while True:
+                line = await stream.read()
+                if line:
+                    callback(line)
+                else:
+                    break
+
+        async def _feed_stream(input):
+            if input:
+                # replicates what proc.communicate() does with stdin
+                await proc._feed_stdin(input)
+
+        await asyncio.wait(
+            [
+                _read_stream(proc.stdout, lambda l: tee(l, stdout, 1)),
+                _read_stream(proc.stderr, lambda l: tee(l, stderr, 2)),
+                _feed_stream(input=stdin)
+            ]
```

Comment:
this is essentially what's done in [asyncio.subprocess](https://github.com/python/cpython/blob/3.11/Lib/asyncio/subprocess.py#L182-L197)